src/library/Connect/WorkloadInformationProvider.cs

```diff
@@ -480,21 +480,33 @@ private int GetServicePortValue(V1ServicePort servicePort, V1Container container
                 var isInWorkloadNamespace = StringComparer.OrdinalIgnoreCase.Equals(endpoint.Metadata.Namespace(), workloadNamespace);
                 foreach (var subset in endpoint.Subsets)
                 {
-                    // For headless service, we only add entries that specify hostname,
-                    // if hostname its not specify there is no dsn to reach the replicate.
-                    var addresses = subset.Addresses?.Where(a => !string.IsNullOrWhiteSpace(a?.Hostname));
-                    if (addresses == null || !addresses.Any())
+                    if (subset.Addresses == null)
                     {
                         continue;
                     }
-
-                    foreach (var address in addresses)
+                    
+                    foreach (var address in  subset.Addresses)
                     {
+                        if (address == null) {
+                            continue;
+                        }
+                        string dns = "";
+                        // If hostname is empty for the address, then dns used is that of the endpoint
+                        if (!string.IsNullOrWhiteSpace(address.Hostname))

```

Comment:
This is suggestion: we can do as a switch statement here instead of if statements as the user suggested ?

Reply:
I decided agains switch because number of conditions is not large and readability seemed better this way. https://www.geeksforgeeks.org/switch-vs-else/ this is some reading I did on switch vs if. It seems switch is preffer if number if else nested cases is large.